coriolisclient/cli/utils.py

```diff
@@ -185,3 +188,34 @@ def get_option_value_from_args(args, option_name, error_on_no_value=True):
             "No '%s[-file]' parameter was provided." % option_arg_name)
 
     return value
+
+
+def compose_user_scripts(global_scripts, instance_scripts):
+    ret = {
+        "global":{},
+        "instances": {}
+    }
+    global_scripts = global_scripts or []
+    instance_scripts = instance_scripts or []
+    for glb in global_scripts:
+        split = glb.split("=", 1)
+        if len(split) != 2:
+            continue
+        if split[0] not in constants.OS_LIST:
+            raise ValueError(
+                "Invalid OS %s. Available options are: %s" % (
+                    split[0], ", ".join(constants.OS_LIST)))
+        if os.path.isfile(split[1]) is False:
+            raise ValueError("Could not find %s" % split[1])
+        with open(split[1]) as sc:
+            ret["global"][split[0]] = sc.read()
```

Comment:
NIT: Do you think it would be worth to add any file encoding checks to ensure it's utf-8/16le for each platform?
There's realistically no way to ensure the encoding on the scripts will work with the workers in advance, so the best we could do IMO is force it to be ascii/utf directly from the client.